src/main/java/com/dyescape/dyescapebot/util/TimeUtil.java

```diff
@@ -0,0 +1,111 @@
+     * returned {@link LocalDateTime} indicates when the punishment ends. The result
+     * time is in UTC timezone.
+     * @param relativeTime
+     * @return {@link LocalDateTime}
+     * @author Dennis van der Veeke
+     */
+    public static long parseFromRelativeString(String relativeTime) {
+        PeriodFormatter formatter = new PeriodFormatterBuilder()
+                .appendDays().appendSuffix("d")
+                .appendDays().appendSuffix("day")
+                .appendDays().appendSuffix("days")
+                .appendHours().appendSuffix("h")
+                .appendHours().appendSuffix("hour")
+                .appendHours().appendSuffix("hours")
+                .appendMinutes().appendSuffix("m")
+                .appendMinutes().appendSuffix("minute")
+                .appendMinutes().appendSuffix("minutes")
+                .toFormatter();
+
+        Period period = formatter.parsePeriod(relativeTime);
+        return period.toStandardDuration().getMillis();
+    }
+
+    /**
+     * Takes a time in milliseconds and parses it to a string which (humanly readable)
+     * parses the time. Includes nice grammar features.
+     * @param milliseconds
+     * @return Parsed punishment time
+     * @author Dennis van der Veeke
+     */
+    public static String parsePunishmentTime(long milliseconds) {
+
+        long secs = milliseconds / 1000;
+        long mins = secs / 60;
+        long hours = mins / 60;
+        long days = hours / 24;
+
+        StringBuilder builder = new StringBuilder();
+
+        if (days > 0) {
+            if (days == 1) {
+                builder.append(String.format("%s day", days));
+            } else {
+                builder.append(String.format("%s days", days));
+            }
+        }
+
+        if (hours % 24 != 0) {
+            if (!builder.toString().isEmpty()) {
+                if (hours % 24 == 1) {
+                    builder.append(String.format(", %s hour", hours % 24));
+                } else {
+                    builder.append(String.format(", %s hours", hours % 24));
+                }
+            } else {
+                if (hours % 24 == 1) {
+                    builder.append(String.format("%s hour", hours % 24));
+                } else {
+                    builder.append(String.format("%s hours", hours % 24));
+                }
+            }
+        }
+
+        if (mins % 60 != 0) {
+            if (!builder.toString().isEmpty()) {
+                if (mins % 60 == 1) {
+                    builder.append(String.format(", %s minute", mins % 60));
+                } else {
+                    builder.append(String.format(", %s minutes", mins % 60));
+                }
+            } else {
+                if (mins % 60 == 1) {
+                    builder.append(String.format("%s minute", mins % 60));
+                } else {
+                    builder.append(String.format("%s minutes", mins % 60));
+                }
+            }
```

Comment:
I'm sure there's a cleaner way to write this. The pluralization especially. A method could be made to simplify it.

Reply:
Simplified it a bit. Got rid of the overload of unnecessairy if statements. Would a method still be nice to have with the new version?